app/gui2/src/stores/imports.ts

```diff
@@ -0,0 +1,589 @@
+      } as RequiredImport,
+      expected: true,
+    },
+    {
+      description: 'Module imported by alias',
+      existing: {
+        from: unwrap(tryQualifiedName('Standard.Base')),
+        imported: { kind: 'Module', alias: unwrap(tryIdentifier('MyBase')) },
+      } as Import,
+      required: {
+        kind: 'Qualified',
+        module: unwrap(tryQualifiedName('Standard.Base')),
+      } as RequiredImport,
+      expected: true,
+    },
+    {
+      description: 'Module imported from parent',
+      existing: {
+        from: unwrap(tryQualifiedName('Standard')),
+        imported: { kind: 'List', names: [unwrap(tryIdentifier('Base'))] },
+      } as Import,
+      required: {
+        kind: 'Qualified',
+        module: unwrap(tryQualifiedName('Standard.Base')),
+      } as RequiredImport,
+      expected: true,
+    },
+    {
+      description: 'Module imported from parent with all',
+      existing: {
+        from: unwrap(tryQualifiedName('Standard')),
+        imported: { kind: 'All', except: [] },
+      } as Import,
+      required: {
+        kind: 'Qualified',
+        module: unwrap(tryQualifiedName('Standard.Base')),
+      } as RequiredImport,
+      expected: true,
+    },
+    {
+      description: 'Module hidden when importing all from parent',
+      existing: {
+        from: unwrap(tryQualifiedName('Standard')),
+        imported: { kind: 'All', except: [unwrap(tryIdentifier('Base'))] },
+      } as Import,
+      required: {
+        kind: 'Qualified',
+        module: unwrap(tryQualifiedName('Standard.Base')),
+      } as RequiredImport,
+      expected: false,
+    },
+    {
+      description: 'Type imported from module by name',
+      existing: {
+        from: unwrap(tryQualifiedName('Standard.Base')),
+        imported: { kind: 'List', names: [unwrap(tryIdentifier('Table'))] },
+      } as Import,
+      required: {
+        kind: 'Unqualified',
+        from: unwrap(tryQualifiedName('Standard.Base')),
+        import: unwrap(tryIdentifier('Table')),
+      } as RequiredImport,
+      expected: true,
+    },
+    {
+      description: 'Type imported from module by all',
+      existing: {
+        from: unwrap(tryQualifiedName('Standard.Base')),
+        imported: { kind: 'All', except: [] },
+      } as Import,
+      required: {
+        kind: 'Unqualified',
+        from: unwrap(tryQualifiedName('Standard.Base')),
+        import: unwrap(tryIdentifier('Table')),
+      } as RequiredImport,
+      expected: true,
+    },
+    {
+      description: 'Type hidden when importing all',
+      existing: {
+        from: unwrap(tryQualifiedName('Standard.Base')),
+        imported: { kind: 'All', except: [unwrap(tryIdentifier('Table'))] },
+      } as Import,
+      required: {
+        kind: 'Unqualified',
+        from: unwrap(tryQualifiedName('Standard.Base')),
+        import: unwrap(tryIdentifier('Table')),
+      } as RequiredImport,
+      expected: false,
+    },
```

Comment:
nitpick: on this list we could have all fields be a string, and do converting/unwrapping in the test.